### ISSUE TITLE
C# namespace imports insertion point before fileheader

### ISSUE THREAD

**Issue by markuskorbel** [dotnet/roslyn#8797]
The current logic for importing new namespaces into C# code (I guess similar for VB but I didn't double check) from the class:

`roslyn/src/Workspaces/CSharp/Portable/Editing/CSharpImportAdder.cs
`
goes roughly like this:
1) Find existing imports and insert before
2) Or if there are none insert in the root

When using stylecop rules though the using statements are nested within the namespace, example:

```
<Current logic inserts here>
namespace MyProject
{
    <But should be inserting here?>
    using System;
}
```

But it also inserts new namespaces before file headers in this scenario since it doesn't find the other imports, example:

```
<Current logic inserts here>
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="File.cs" company="MyOrgnaization">
// Copyright (C) MyOrgnaization 2016
// </copyright>
// --------------------------------------------------------------------------------------------------------------------
<If imports inside namespace are problematic, the at least skip the file header(s) and import here>
namespace MyProject
{
    <But should be ideally inserting here?>
    using System;
}
```

**Comment by Pilchie** [comment]
VB doesn't allow `Imports` statements inside namespaces.  I don't think we'd ever change the default for C# either, but we might consider making an option.

However, we *should* place it after the banner text, and the fix should be similar to the one for #7117.

**Comment by markuskorbel** [comment]
Makes sense, god it's been long since I wrote something in VB ... forgot about the imports :)
